google-cloud-datastore/src/main/java/com/google/cloud/datastore/AggregationResult.java

```diff
@@ -24,21 +27,52 @@
 /** Represents a result of an {@link AggregationQuery} query submission. */
 public class AggregationResult {
 
-  private final Map<String, LongValue> properties;
+  private final Map<String, Value<?>> properties;
 
-  public AggregationResult(Map<String, LongValue> properties) {
+  public AggregationResult(Map<String, Value<?>> properties) {
```

Comment:
Are we certain that this is not a breaking change?
Maybe to be safe, just add another constructor
`public AggregationResult(Map<String, DoubleValue> properties)`?
cc: @burkedavison for a 2nd opinion.

Reply:
Re: another constructor
You can't have both `public AggregationResult(Map<A, B> map)` and `public AggregationResult(Map<A, C> map)`, as their signatures overlap (`public AggregationResult(Map map)` once type erasure occurs. It will throw a compiler error if you give it a try.

Re: breaking change
```java
Map<String, LongValue> longMap = Collections.emptyMap();
new AggregationResult(longMap); // This will also create a compiler error with the current syntax.
```
Instead, use:
```java
public AggregationResult(Map<String, ? extends Value<?>> properties)
```
and then the compiler will be happy.

Reply:
Thanks!

Reply:
Thanks for pointing it out, have addressed this in [b60995b](https://github.com/googleapis/java-datastore/pull/1067/commits/b60995b0be6813d5f302006b02ae74729d0aa33f).